src/wasm.rs

```diff
@@ -223,3 +223,8 @@ pub fn platform() -> Platform {
         Platform::Unknown(string.to_string())
     }
 }
+
+#[inline(always)]
+pub fn arch() -> Arch {
+    ARCH.parse().unwrap()
```

Comment:
This should probably just be an `if` statement checking `cfg(target_pointer_width = 64)`

Reply:
It seems that the following impl won't work, at least for `asmjs-unknown-emscripten`:

```rust
pub fn arch() -> Arch {
    println!("ARCH: {}", std::env::consts::ARCH);
    if cfg!(target_poiner_width = "32") {
        Arch::Wasm32
    } else if cfg!(target_pointer_width = "64"){
        Arch::Wasm64
    } else {
        unreachable!("Arches other than wasm32 and wasm64 are not supported")
    }
}
```

This is weird

```shell
$ cargo b --target asmjs-unknown-emscripten -q

$ node target/asmjs-unknown-emscripten/debug/rust.js
ARCH: wasm32
thread 'main' panicked at 'internal error: entered unreachable code: Arches other than wasm32 and wasm64 are not supported', /home/steve/Documents/workspace/whoami/src/wasm.rs:234:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
BTW, see the output of `std::env::consts::ARCH` on `asmjs-unknown-emscripten`, it is `wasm32`, not `asmjs`, then seems that `Arch::AsmJs` will never be constructed

Reply:
It looks like a typo, you wrote `target_pointer_width` as `target_poiner_width` (missing the 't').

We should remove `AsmJs` variant, then.

Reply:
Sorry for being so careless:(

Got it!